monai/transforms/utils.py

```diff
@@ -1676,14 +1677,14 @@ def scale_affine(affine, spatial_size, new_spatial_size, centered: bool = True):
         Scaled affine matrix.
 
     """
+    r = int(spatial_rank)
     if spatial_size == new_spatial_size:
-        return affine
-    r = len(affine) - 1
+        return np.eye(r + 1)
```

Comment:
here should we add a check to ensure the value `spatial_rank ` matches `spatial_size` and `new_spatial_size`?

Reply:
It seems spatial rank is redundant... We can infer it from len(new_spatial_size), let's remove the spatial_rank arg

Reply:
Another question here, if we don't pass `affine` in this function, does it mean we only record something like "affine_scale_ratio" in the `TraceableTransform.track_transform_meta`? Am I understand correctly?

Reply:
Correct, the result is then combined with the original affine in https://github.com/Project-MONAI/MONAI/blob/59f42f234bf04e8d001fb77d8813e7efd23d9bd9/monai/transforms/inverse.py#L182

Reply:
removed `spatial_rank` from `scale_affine`, now it's computed from `r = max(len(new_spatial_size), len(spatial_size))`, so that when it's 2d->3d or 3d->2d can still work https://github.com/Project-MONAI/MONAI/pull/5860/commits/1957962ff1edf480f6d8692224c7a576cdabee27